components/shared_code/src/shared_data_model/sources/github.py

```diff
@@ -0,0 +1,102 @@
+"""GitHub source."""
+# Author: Tobias Termeczky
+# Company: the/experts.
+
+from pydantic import HttpUrl
+
+from shared_data_model.meta.entity import (
+    Entity,
+    EntityAttribute,
+    EntityAttributeType,
+)
+from shared_data_model.meta.source import Source
+from shared_data_model.parameters import (
+    URL,
+    MergeRequestState,
+    MultipleChoiceParameter,
+    PrivateToken,
+    StringParameter,
+    TargetBranchesToInclude,
+)
+
+ALL_GITHUB_METRICS = [
+    "merge_requests",
+]
+
+GITHUB_BRANCH_HELP_URL = HttpUrl(
+    "https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository"
+)
+
+GITHUB = Source(
+    name="GitHub",
+    description="GitHub provides a cloud-based hosting service for Git repositories, offering tools for version control, source code management, pull requests, issue tracking, and continuous integration/continuous deployment (CI/CD).",
+    url=HttpUrl("https://github.com/about"),
+    parameters={
+        "url": URL(
+            name="GitHub instance URL",
+            help="URL of the GitHub instance, with port if necessary, but without path. For example, "
+            "'https://github.com'.",
+            validate_on=["private_token"],
+            metrics=ALL_GITHUB_METRICS,
+        ),
+        "owner": StringParameter(
+            name="Owner (name of owner of the repository)",
+            short_name="owner",
+            mandatory=True,
+            help_url=HttpUrl("https://docs.github.com/en/account-and-profile"),
+            metrics=ALL_GITHUB_METRICS,
+        ),
+        "repository": StringParameter(
+            name="Repository (name of the repository)",
+            short_name="repository",
+            help_url=HttpUrl("https://docs.github.com/en/repositories/creating-and-managing-repositories"),
+            mandatory=True,
+            metrics=ALL_GITHUB_METRICS,
+        ),
+        "private_token": PrivateToken(
+            name="Fine-Grained Personal Access Tokens (with Read scope of the repository)",
+            short_name="Fine-Grained Personal Access Tokens",
+            help_url=HttpUrl(
+                "https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens"
+            ),
+            metrics=ALL_GITHUB_METRICS,
+        ),
+        "merge_request_state": MergeRequestState(
+            name="Pull request state",
+            short_name="PR State",
+            values=["Open", "Merged", "Closed"],
+            api_values={"Open": "OPEN", "Merged": "MERGED", "Closed": "CLOSED"},
+        ),
+        "review_decision": MultipleChoiceParameter(
+            name="Review Decision",
```

Comment:
Why the capital D? It's English not German ;-)

```suggestion
            name="Review decision",
```